pkg/cache/context.go

```diff
@@ -765,3 +767,62 @@ func (ctx *Context) SchedulerNodeEventHandler() func(obj interface{}) {
 	// this is not required in some tests
 	return nil
 }
+
+func createConfigMapSelector() (labels.Selector, error) {
+	slt := labels.NewSelector()
+	appVals := [] string {"yunikorn"}
+	req, err := labels.NewRequirement(common.LabelApp, selection.Equals, appVals)
+	if err != nil {
+		return  nil, err
+	}
+	slt = slt.Add(*req)
+	return slt, nil
+}
+
+func findYKConfigMap(configMaps []*v1.ConfigMap) (*v1.ConfigMap, error) {
+	var ykconf *v1.ConfigMap
+	foundConf := false
+	if len(configMaps) == 0 {
+		return nil, fmt.Errorf("configmap with label app:yunikorn not found")
+	}
+	for _, c := range configMaps {
+		if c.Name == common.DefaultConfigMapName {
+			ykconf = c
+			foundConf = true
+			break
+		}
+	}
+	if !foundConf {
+		return nil, fmt.Errorf("configmap with name %s not found", common.DefaultConfigMapName)
+	}
+	return ykconf, nil
+}
+/*
+Save the configmap and returns the old one and an error if the process failed
+ */
+func (ctx *Context) SaveConfigmap(data string) (string, error) {
+	slt, err := createConfigMapSelector()
+	if err != nil {
+		return "", err
+	}
+
+	configMaps, err := ctx.apiProvider.GetAPIs().ConfigMapInformer.Lister().List(slt)
+	if err != nil {
+		return "", err
+	}
+	ykconf, err := findYKConfigMap(configMaps)
+	if err != nil {
+		return "", err
+	}
+
+	newConfData := map[string] string {"queues.yaml":strings.ReplaceAll(data, "\r\n", "\n")}
```

Comment:
not sure why the replaceAll is needed here. 
Why we have `\r\n` in the fist place?

Reply:
When I did some testing using Advanced REST client, without this replace, the saved configuration in the Kubernetes dashboard wasn't parsed correctly.  Instead of breaking the lines, \r\n was showed. By replacing "\r\n'' with "\n", the new configuration will be shown in the similar way, as before this feature.